cmd/deploy_test.go

```diff
@@ -1018,3 +1019,60 @@ func TestDeploy_NamespaceRedeployWarning(t *testing.T) {
 		t.Fatalf("expected function to be updated with namespace 'myns'.  got '%v'", f.Namespace)
 	}
 }
+
+// TestDeploy_NamespaceRedeployNoWarning ensures that the --remote flag is
+// persisted in the global config.
+func TestDeploy_persistRemoteFlag(t *testing.T) {
+	root, rm := Mktemp(t)
+	defer rm()
+
+	// A function with minimum required values for deployment
+	f := fn.Function{
+		Root:     root,
+		Name:     "myfunc",
+		Runtime:  "go",
+		Registry: "example.com/alice",
+	}
+	if err := fn.New().Create(f); err != nil {
+		t.Fatal(err)
+	}
+
+	// Deploy using an instance of the deploy command which uses a fully default
+	// (noop filled) Client.  Execution should complete without error.
+	cmd := NewDeployCmd(NewClientFactory(func() *fn.Client {
+		return fn.New()
+	}))
+
+	// Set the --remote flag
+	cmd.SetArgs([]string{"--git-url=https://git.hub.com/alice/myfunc", "--remote"})
+	if err := cmd.Execute(); err != nil {
+		t.Fatal(err)
+	}
+
+	// Ensure that successfully executing the command results in the --remote
+	// flag being persisted. Load the global config and check the value.
+	cfg, err := config.NewDefault()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !cfg.Remote {
+		t.Fatal("expected remote flag to be persisted as true in global config")
+	}
```

Comment:
I really like where this PR is headed:  that we can use the global config to define a default (globally) for the --remote flag.

This is, however, a separate feature from _persisting_ the value of the --remote flag (for an individual Function), because that setting would be written to `.func`.

The PR I am working on now actually injects the current Function context into the final value returned when instantiating that global config object.  That's going to be a little bit tricky because this is the first function-scoped value which is not actually part of the core (it pertains only to the `cmd` package).

I might consider using THIS PR to deliver the global config remote value setting, and have this test confirm it is read (perhaps citing a new issue that the global config should support configuring the global default "remote" setting).